front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerHeaderFields.test.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import BannerHeaderFields from './BannerHeaderFields';
+import { render, screen, waitFor, fireEvent } from 'utils/testUtils/rtl';
+
+jest.mock('utils/cl-intl');
+jest.mock('hooks/useAppConfigurationLocales', () =>
+  jest.fn(() => ['en', 'nl-NL'])
+);
+
+describe('BannerHeaderFields', () => {
+  it('renders with correctly displayed text', () => {
+    render(
+      <BannerHeaderFields
+        bannerHeaderMultiloc={{ en: 'banner header' }}
+        bannerSubheaderMultiloc={{ en: 'banner subheader' }}
+        onHeaderChange={jest.fn()}
+        onSubheaderChange={jest.fn()}
+        title={'banner header fields'}
+        inputLabelText={'header input label text'}
+        subheaderInputLabelText={'subheader input label text'}
+      />
+    );
+
+    expect(screen.getByText('banner header fields')).toBeInTheDocument();
+    expect(screen.getByDisplayValue('banner subheader')).toBeInTheDocument();
+  });
+
+  it('calls onChange handlers properly', async () => {
+    const onHeaderChange = jest.fn();
+    const onSubheaderChange = jest.fn();
+
+    render(
+      <BannerHeaderFields
+        bannerHeaderMultiloc={{ en: 'banner header' }}
+        bannerSubheaderMultiloc={{ en: 'banner subheader' }}
+        onHeaderChange={onHeaderChange}
+        onSubheaderChange={onSubheaderChange}
+        title={'banner header fields'}
+        inputLabelText={'header input label text'}
+        subheaderInputLabelText={'subheader input label text'}
+      />
+    );
+
+    await waitFor(() => {
+      fireEvent.change(screen.getByDisplayValue('banner header'), {
+        target: {
+          value: 'new banner header',
+        },
+      });
+    });
+
+    fireEvent.change(screen.getByDisplayValue('banner subheader'), {
+      target: {
+        value: 'new banner subheader',
+      },
+    });
+
+    expect(onHeaderChange).toHaveBeenCalledWith({ en: 'new banner header' });
+    expect(onSubheaderChange).toHaveBeenCalledWith({
+      en: 'new banner subheader',
+    });
+  });
+
+  it('does not accept too-long strings', async () => {
```

Comment:
How does this test if the strings are not too long?

Reply:
That seems like a peculiar way to test strings that are not too long. :D Could you add a comment to clarify this?

Reply:
cc @IvaKop

Reply:
Without having enough context here - perhaps the input field should have max length set to the character limit? It's not clear to me why onChange is not called for longer strings but they are allowed to be set in the field?

Reply:
Sorry, was just trying to quickly cover all the cases I saw present in this component. I didn't actually build the component, was just asked to help out in testing, so it seemed like a decent test. On deeper examination it looks like the `maxCharCount` property goes into our component library's `InputMultilocWithLocaleSwitcher` so this behavior should more accurately be tested there. Removing the test now